### PR TITLE
Calculate number of nodes in the tree estimators

### PR DESCRIPTION
**Pull request by avolkov-intel**
## Description

- Automatically report total number of tree nodes (`n_nodes`) for Random Forest and Extra Trees estimators after training
- Register `n_nodes` as an indifferent metric in the report so it's compared across libraries rather than treated as a grouping parameter

---

<details>
<summary>Checklist:</summary>

**Completeness and readability**

- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have updated the documentation to reflect the changes or created a separate PR with updates and provided its number in the description, if necessary.
- [ ] Git commit message contains an appropriate signed-off-by string _(see [CONTRIBUTING.md](https://github.com/uxlfoundation/scikit-learn-intelex/blob/main/CONTRIBUTING.md#pull-requests) for details)_.
- [ ] I have resolved any merge conflicts that might occur with the base branch.

**Testing**

- [ ] I have run it locally and tested the changes extensively.
- [ ] All CI jobs are green or I have provided justification why they aren't.
- [ ] I have extended testing suite if new functionality was introduced in this PR.

</details>

### CODE REVIEW

File: sklbench/benchmarks/sklearn_estimator.py [hunk 1]
```diff
@@ -134,6 +134,16 @@ def get_subset_metrics_of_estimator(
                 and isinstance(iterations[0], Union[Numeric, NumpyNumeric].__args__)
             ):
                 metrics.update({"iterations": int(iterations[0])})
+        if hasattr(estimator_instance, "estimators_"):
```
Comment (david-cortes-intel) [inline sklbench/benchmarks/sklearn_estimator.py:137]: This will also be set in other types of estimators - for example:
https://scikit-learn.org/stable/modules/generated/sklearn.multiclass.OneVsRestClassifier.html

Perhaps it could check if the first entry there is of tree type.

Also what about cuML? I guess it could be done by calling `as_sklearn()`:
https://docs.rapids.ai/api/cuml/nightly/pickling_cuml_models/#Converting-Between-cuML-and-scikit-learn-Models
